src/app/common/grade/grade.component.ts

```diff
@@ -12,14 +13,19 @@ export class GradeComponent implements OnInit {
   @Input('modifier') showModifier: boolean = false;
   @Input() legacy: boolean = false;
 
-  grade: Grade;
+  // grade: Grade;
+  grade: IGrade;
   modifier: number = 0;
 
   gradeLetters = ['a', 'a+', 'b', 'b+', 'c', 'c+'];
 
-  constructor() {}
+  constructor(private GradingSystemsService: GradingSystemsService) {}
 
   ngOnInit(): void {
-    this.grade = new Grade(this.difficulty);
+    // this.grade = new Grade(this.difficulty);
```

Comment:
Uncomment this and comment out `this.grade = name;` to see the old implementation if you want to quickly compare the differences.

And then also uncomment the import and property definition above.